skepticoin/merkletree.py

```diff
@@ -32,54 +38,59 @@ def hash(self):
 
         return sha256d(b"".join(c.hash() for c in self.children))
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "M(%s, (%s))" % (human(self.hash())[:7], self.children)
 
 
-def _get_merkle_tree(list_of_nodes):
+def _get_merkle_tree(list_of_nodes: List[MerkleNode]) -> MerkleNode:
     if len(list_of_nodes) == 1:
         return list_of_nodes[0]
 
     new_list = []
     for chunk in _chunks(list_of_nodes, 2):
         if len(chunk) == 2:
-            new_list.append(MerkleNode(chunk[0].index, (chunk[0], chunk[1])))
+            new_list.append(MerkleNode(chunk[0].index, [chunk[0], chunk[1]]))
         else:  # implied: len(chunk) == 1
             new_list.append(chunk[0])
 
     return _get_merkle_tree(new_list)
 
 
-def get_merkle_tree(list_of_hashes):
+def get_merkle_tree(list_of_hashes: List[bytes]) -> MerkleNode:
     return _get_merkle_tree(
-        [MerkleNode(i, (), h) for (i, h) in enumerate(list_of_hashes)]
+        [MerkleNode(i, [], h) for (i, h) in enumerate(list_of_hashes)]
     )
 
 
-def get_proof(merkle_node, index_of_interest):
+def get_proof(merkle_node: MerkleNode, index_of_interest: int) -> MerkleNode:
     if not merkle_node.children:
         return merkle_node
 
+    reconstruct: Callable[[MerkleNode, MerkleNode], List[MerkleNode]]
+
     # our nodes always have either 0 or 2 children, never 1
     if index_of_interest >= merkle_node.children[1].index:
         other, recurse_into = merkle_node.children
-        reconstruct = lambda ot, rec: (ot, rec)  # noqa
+        reconstruct = lambda ot, rec: [ot, rec]
```

Comment:
the `noqa` directive was presumably to get flake8 to shut up about inline lambda's; I'm not sure how you managed to do that if you've removed that (if at all)